src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea() : number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (!this.checkTriangle()) {
+      throw new Error('There are no correct triangle sizes');
+    }
+  }
+
+  checkTriangle(): boolean {
+    const maxSize = Math.max(this.a, this.b, this.c);
+
+    if ((maxSize < this.a + this.b + this.c - maxSize)
+      && this.a > 0
+      && this.b > 0
+      && this.c > 0) {
```

Comment:
Let's store these conditions in the array and check it with the `some` method

Reply:
Fixed. Its a good way to check conditions. Thanks.